### PR TITLE
Number of fixes for ACL commands

### LINKED ISSUE #1283: Delete command should print out rules that were deleted on the backend
## Description

Delete command should print out rules that were deleted on the backend

### PR DESCRIPTION
**Pull request by wtrocki**
Closes #1283 #1282 #1284 

I have put all fixes in single PR to minimize effort/hassle with reviews.
All commands can be reviewed and verified using the same 2 commands provided in the verification file.

### CODE REVIEW

File: pkg/cmd/kafka/acl/list/list.go [hunk 1]
```diff
@@ -154,7 +154,7 @@ func runList(opts *options) (err error) {
 	case dump.EmptyFormat:
 		opts.logger.Info("")
 		permissions := permissionsData.GetItems()
-		rows := aclutil.MapACLsToTableRows(permissions, opts.localizer)
+		rows := aclutil.MapACLsToTableRows(&permissions, opts.localizer)
```
Comment (craicoverflow) [inline pkg/cmd/kafka/acl/list/list.go:157]: Unnecessary IMO, they are not that large, and it makes it a bit harder to read.

File: pkg/kafka/aclutil/acl_table.go [hunk 2]
```diff
@@ -16,10 +16,10 @@ type permissionsRow struct {
 }
 
 // MapACLsToTableRows converts a list of ACL bindings into a formatted table for printing
-func MapACLsToTableRows(bindings []kafkainstanceclient.AclBinding, localizer localize.Localizer) []permissionsRow {
-	rows := make([]permissionsRow, len(bindings))
+func MapACLsToTableRows(bindings *[]kafkainstanceclient.AclBinding, localizer localize.Localizer) []permissionsRow {
```
Comment (craicoverflow) [inline pkg/kafka/aclutil/acl_table.go:19]: A slice wraps an array, and holds a reference to that array already, so an extra reference is unnecessary:

https://golang.org/doc/effective_go#slices

**Comment by craicoverflow** [comment]
Screenshot 2021-11-05 at 12 11 27

#1282 is not being resolved.

**Review by craicoverflow — CHANGES_REQUESTED** [review]
#1282 does not seem to be fixed yet. The rest is good.

**Comment by wtrocki** [comment]
```
rhoas kafka acl create --all-accounts --permission allow --operation all --topic all  -y
The following ACL rule is going to be created:

  PRINCIPAL      PERMISSION   OPERATION   DESCRIPTION   
 -------------- ------------ ----------- -------------- 
  All accounts   ALLOW        ALL         TOPIC is "*"  


⣻ Adding ACL to Kafka instance "test" 
```

Done

**Comment by wtrocki** [comment]
Closes #1283 #1282 #1284

**Comment by wtrocki** [comment]
I canot figure out how to generate documentation. Locally documentation has no changes but CI does show them.

**Comment by craicoverflow** [comment]
> I canot figure out how to generate documentation. Locally documentation has no changes but CI does show them.

CI checks against `main`. Your local branch is behind main.

**Comment by craicoverflow** [comment]
Approving assuming docs will be fixed before merging.

**Pull request merged**